### ISSUE TITLE
Trying to implement in tfjs... can anyone explain this bit of code?

### ISSUE THREAD

**Issue by talvasconcelos** [edwardhdlu/q-trader#8]
I'm trying to implement this in tensorflow js but i'm running into some issue with these lines in agent.py:

```
target_f = self.model.predict(state)
target_f[0][action] = target
self.model.fit(state, target_f, epochs=1, verbose=0)
```

What does target_f means? final target? 

What is target[action] ? Why are we setting target_f to the predict and then assingning it something else?

thanks, 
Tiago

**Comment by kkuette** [comment]
As i remember, target_f is considered as a label for self.model.fit()
target_f is an array of three columns, and it assign the reward to the action he did in the first row.

**Comment by talvasconcelos** [comment]
I drilled it down after all. In tensorflow js, in order to get the array output from the predict, you need to pass .dataSync()... after a couple of console.logs i figured it out but, thanks for the explanation.
